### DISCUSSION TITLE
Failure installing Laminas LDAP

### DISCUSSION THREAD

**Discussion by sebelk** [kimai/kimai#2558] [On-Premise: Installation & Updates]
**To Reproduce**
1. Run the command `composer require laminas/laminas-ldap --no-dev --optimize-autoloader` as is documented at https://www.kimai.org/documentation/ldap.html#installation
1. See error

```
[Symfony\Component\Console\Exception\RuntimeException]  
  The "--no-dev" option does not exist.                   
                                                          

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-scripts] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--no-unpack] [--unpack] [--] [<packages>]...
```

**Additional context**
 - Kimai version: 1.14.1 
 - PHP version: 7.3
 - OS: Debian Buster

I've taken a look at https://getcomposer.org/doc/03-cli.md#require and says nothing about --no-dev option. 

Also somone posted this same issue:  https://github.com/composer/composer/issues/6303

Should I replace require by **install** in that command?

Thanks in advance!

**Comment by kevinpapst** [comment]
Try `composer require laminas/laminas-ldap --optimize-autoloader `

**Reply by sebelk** [reply]
Thanks @kevinpapst. It worked

One question more:

after run that command it suggest that execute:
`npm install --force`

and it outputs:

```
npm WARN using --force I sure hope you know what you are doing.
npm WARN friendly-errors-webpack-plugin@2.0.0-beta.2 requires a peer of webpack@^4.0.0 but none is installed. You must install peer dependencies yourself.

audited 1059 packages in 23.649s

108 packages are looking for funding
  run `npm fund` for details

found 1 moderate severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details

```
Does make sense to install that dependency?

Thanks in advance!
Sergio

**Reply by kevinpapst** [reply]
No need to execute these steps. Everything thats required is in the documentation.

**Comment by sebelk** [comment]
Thanks Kevin, anyway I couldn't restrain myself and execute these steps :shrug: 
Everything is  working fine.
